xml/chapter5/section5/subsection2.xml

```diff
@@ -1117,59 +1144,62 @@ function compile_lambda_body(exp, fun_entry) {
-    To ensure that every function ends by executing a return statement, <JAVASCRIPTINLINE>compile_lambda_body</JAVASCRIPTINLINE>
-    appends a return statement whose return expression is the literal <JAVASCRIPTINLINE>undefined</JAVASCRIPTINLINE>
-    to the lambda body. Because return statements always use the <JAVASCRIPTINLINE>"return"</JAVASCRIPTINLINE> linkage
-    descriptor for their return expressions, all function bodies will end with a <JAVASCRIPTINLINE>"return"</JAVASCRIPTINLINE>
-    linkage<EMDASH/>and not the <JAVASCRIPTINLINE>"next"</JAVASCRIPTINLINE> we passed in above.<FOOTNOTE>Note
-    that a sequence of statements compiled with linkage descriptor <EM>l</EM> will compile all its statements with
-  linkage <JAVASCRIPTINLINE>"next"</JAVASCRIPTINLINE> except the last, which uses <EM>l</EM>.</FOOTNOTE>
-    
-    <SNIPPET EVAL="no">
-      <NAME>append_return_undefined</NAME>
-      <JAVASCRIPT>
+  <SPLIT>
+    <JAVASCRIPT>
+      <TEXT>
+	To ensure that every function ends by executing a return statement,
+	<JAVASCRIPTINLINE>compile_lambda_body</JAVASCRIPTINLINE>
+	appends a return statement whose return expression is the literal
+	<JAVASCRIPTINLINE>undefined</JAVASCRIPTINLINE>
+	to the lambda body. Because return statements always use the
+	<JAVASCRIPTINLINE>"return"</JAVASCRIPTINLINE> linkage
+	descriptor for their return expressions, all function bodies will end with a
+	<JAVASCRIPTINLINE>"return"</JAVASCRIPTINLINE>
+	linkage<EMDASH/>and not the <JAVASCRIPTINLINE>"next"</JAVASCRIPTINLINE> we
+	passed as argument to <JAVASCRIPTINLINE>compile</JAVASCRIPTINLINE> in
+	<JAVASCRIPTINLINE>compile_lambda_body</JAVASCRIPTINLINE>.<FOOTNOTE>Note
+	that a sequence of statements compiled with a linkage descriptor <META>linkage</META>
+	uses the
+	linkage <JAVASCRIPTINLINE>"next"</JAVASCRIPTINLINE> for all its
+	component statements except the last, for which it uses the descriptor
+	<META>linkage</META>.</FOOTNOTE>
+	<SNIPPET EVAL="no">
+	  <NAME>append_return_undefined</NAME>
+	  <JAVASCRIPT>
 function append_return_undefined(body) {
-    return list("sequence", list(body, 
-                                 list("return_statement", 
+    return list("sequence", list(body,
+                                 list("return_statement",
                                       list("literal", undefined))));
 }
-      </JAVASCRIPT>
-    </SNIPPET>
-    This simple transformation of lambda bodies represents a third way to ensure that a function that does not
-    return explicitly has the return value <JAVASCRIPTINLINE>undefined</JAVASCRIPTINLINE>.
-    
-    In the metacircular evaluator, we used a return-value object, which also played a
-    role in stopping the sequence evaluation.
-
-    In the explicit-control evaluator, functions that did not return explicitly continued
-    to an entry point that stored <JAVASCRIPTINLINE>undefined</JAVASCRIPTINLINE>
-    in <JAVASCRIPTINLINE>val</JAVASCRIPTINLINE>.
-
-    Source-level transformations like
```

Comment:
removed because our derived forms of 5.1.2 can be seen as source level transformations, and they happen to take place in the interpreter (and in the compiler). I don't think the removal is a big loss.